fenix/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -140,6 +142,25 @@ class IntentReceiverActivity : Activity() {
             }
         }
     }
+
+    private fun tryGettingReferrer(): Uri? {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP_MR1) {
+            return null
+        }
+
+        return try {
+            referrer
+        } catch (e: RuntimeException) {
+            try {
+                referrer
+            } catch (e: Exception) {
+                null
+            }
+        } catch(e: Exception) {
+            null
+        }
+    }
```

Comment:
For some reason, the second access does not cause a `RuntimeException` for the `referrer` access. It does not seem to be timing based, as waiting with a break point does not prevent the error from happening. Inspecting the variable does however throw the first time as well, but not the second time. 

However this whole behavior might be very specific to Spotify's implementation and I'm not sure how important the referrer information is. It seems to work fine without it, so a less complex fix is a simple try catch around the `referrer` access.